futures-core/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+//! Core traits and types for asynchronous operations in Rust.
+
+#![no_std]
+#![deny(missing_docs, missing_debug_implementations)]
+#![doc(html_root_url = "https://docs.rs/futures/0.2")]
+
+#[macro_use]
+#[cfg(feature = "std")]
+extern crate std;
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! if_std {
+    ($($i:item)*) => ($(
+        #[cfg(feature = "std")]
+        $i
+    )*)
+}
+
+#[macro_use]
+mod poll;
+pub use poll::{Async, Poll};
+
+pub mod future;
+pub use future::{Future, IntoFuture};
```

Comment:
This makes me think, could we avoid this for now?

Maybe each crate could have its own prelude which slowly builds over time?

Reply:
Discussed on IRC-- decision was to keep individual preludes for crates and not to reexport items from dependencies.